### ISSUE TITLE
PANIC: called `Result::unwrap()` on an `Err` value: Any { .. } in query-engine/connectors/sql-query-connector/src/error.rs:58:51

### ISSUE THREAD

**Issue by sergiotapia** [prisma/prisma#18122]
Hi Prisma Team! My Prisma Client just crashed. This is the report:
## Versions

| Name            | Version            |
|-----------------|--------------------|
| Node            | v16.15.0           | 
| OS              | darwin-arm64       |
| Prisma Client   | 3.11.0             |
| Query Engine    | 0.1.0              |
| Database        | postgresql         |

## Logs
```
  prisma:tryLoadEnv Environment variables loaded from /Users/sergiotapia/Work/back_end/.env  
  prisma:tryLoadEnv Environment variables loaded from /Users/sergiotapia/Work/back_end/.env  +1s
  prisma:client clientVersion: 3.11.0  
  prisma:client clientEngineType: library  
  prisma:client:libraryEngine internalSetup  
  prisma:tryLoadEnv Environment variables loaded from /Users/sergiotapia/Work/back_end/.env  +33s
  prisma:client clientVersion: 3.11.0  +33s
  prisma:client clientEngineType: library  
  prisma:client:libraryEngine internalSetup  +33s
  prisma:client:libraryEngine Searching for Query Engine Library in /Users/sergiotapia/Work/back_end/node_modules/.prisma/client  +2s
  prisma:client:libraryEngine Searching for Query Engine Library in /Users/sergiotapia/Work/back_end/node_modules/.prisma/client  
  prisma:client:libraryEngine loadEngine using /Users/sergiotapia/Work/back_end/node_modules/.prisma/client/libquery_engine-darwin-arm64.dylib.node  
  prisma:client:libraryEngine loadEngine using /Users/sergiotapia/Work/back_end/node_modules/.prisma/client/libquery_engine-darwin-arm64.dylib.node  
  prisma:client:libraryEngine sending request, this.libraryStarted: false  +3s
  prisma:client:libraryEngine library starting  
  prisma:client:libraryEngine library started  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
  prisma:client prisma.queryRaw(
  prisma:client       SELECT users.id 
  prisma:client       FROM users
  prisma:client       INNER JOIN plans ON users.id = plans.user_id
  prisma:client       WHERE (first_name = $1 OR SPLIT_PART(first_name, ' ', 1) = $2)
  prisma:client       AND last_name = $3
  prisma:client       AND plans.carrier_id IN ($4)
  prisma:client       AND plans.plan_type IN ($5)
  prisma:client       LIMIT 1
  prisma:client     , ["John","John","Doe",["11111111-451f-461b-9a6f-dfb19dada9f9"],["PDP"]])  +5s
  prisma:client Prisma Client call:  
  prisma:client:libraryEngine sending request, this.libraryStarted: true  
```

## Client Snippet
```ts
sync maybeMatchCarrierPlanTypeFirstNameLastName(planType, firstName, lastName): Promise<User | null> {
    const carrierIds = await this.getCarrierIds();

    // Find all the users where the first_name equals the firstName
    // or the SPLIT_PART(first_name, ' ', 1) equals the firstName. Only
    // match against users that have plans under this carrier and plan type.
    const potentialUserId: string = await prisma.$queryRaw`
      SELECT users.id 
      FROM users
      INNER JOIN plans ON users.id = plans.user_id
      WHERE (first_name = ${firstName} OR SPLIT_PART(first_name, ' ', 1) = ${firstName})
      AND last_name = ${lastName}
      AND plans.carrier_id IN (${carrierIds})
      AND plans.plan_type IN (${this.planAliases(planType)})
      LIMIT 1
    `;

    try {
      return await prisma.user.findFirst({
        where: {
          id: { equals: potentialUserId },
        },
        include: {
          plans: {
            where: {
              carrier_id: { in: carrierIds },
              plan_type: { in: this.planAliases(planType) },
            },
            orderBy: { created_at: "desc" },
            take: 1,
          },
        },
      });
    } catch (e) {
      // console.log(e);
      return null;
    }
  }
```

## Schema
I can't share this but it's a very simple User has many Plans. Plans belongs to Carrier.

## Prisma Engine Query
```
{"X":{}}
```

**Comment by sergiotapia** [comment]
I fixed this by using `Prisma.join`:

```
// BAD:
AND plans.carrier_id IN (${carrierIds})

// GOOD:
AND plans.carrier_id IN (${Prisma.join(carrierIds)})

```

**Comment by janpio** [comment]
This should not crash without that - we should take a look.

**Comment by Weakky** [comment]
Hey,

This problem is fixed since 3.14.0 [by this PR](https://github.com/prisma/prisma-engines/pull/2902). The issue is likely that you're trying to compare an array with a column of type `text`.

Versions >= 3.14.0 now print this error:

```
Conversion failed: Couldn't serialize value `Some([Text(Some("ACTIVE"))])` into a `text`. Value is a list but `text` is not.`
```

Thanks for reporting, I'm closing now 🙏
